src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -48,7 +48,7 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
 
                 _pendingProgress = null;
 
-                if (ExperimentalFeature.IsEnabled(ExperimentalFeature.PSAnsiProgressFeatureName) && PSStyle.Instance.Progress.UseOSCIndicator)
+                if (SupportsVirtualTerminal && ExperimentalFeature.IsEnabled(ExperimentalFeature.PSAnsiProgressFeatureName) && PSStyle.Instance.Progress.UseOSCIndicator)
```

Comment:
I wonder why do not check SupportsVirtualTerminal in time we create PSStyle.Instance? It would seem be more generic.

Reply:
Yeah the way we decide within the methods what behaviour to have here isn't great design and while this change seems ok, I think we need to register the need to do this better.

An alternative perspective on how the code should be structured: `PSHostUserInterface` is an abstract class intended to be overridden with the implementation to actually provide. The right object-oriented approach would be to build the user interface object for the given settings, with shared functionality implemented in a common base class.